src/insexpand.c

```diff
@@ -4626,7 +4626,7 @@ get_normal_compl_info(char_u *line, int startcol, colnr_T curs_col)
 
     compl_patternlen = STRLEN(compl_pattern);
 
-    if ((get_cot_flags() & COT_FUZZYCOLLECT) != 0)
+    if ((get_cot_flags() & COT_FUZZYCOLLECT) != 0 && compl_length > 0)
     {
 	// Adjust size to avoid buffer overflow
 	fuzzy_len = (size_t)compl_length * 5 + 10;
```

Comment:
Nit: you can simplify the code and get rid of this guess by using a `garray_T` to build the string.

Reply:
Speaking of which, Neovim has Stringbuilder, maybe Vim can have a wrapper too :)